airflow/models/taskinstance.py

```diff
@@ -194,6 +195,11 @@ def __init__(self, task, execution_date, state=None):
 
         self.queue = task.queue
         self.pool = task.pool
+        if hasattr(task, 'pool_capacity'):
+            self.pool_capacity = task.pool_capacity
+            if task.pool_capacity < 1:
```

Comment:
This check should be done when creating/setting it on the Task/operator, not here.

Reply:
moved the check to baseoperator